docs/src/pages/components/menus/CascadingMenu.js

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import SubMenu from '@material-ui/core/SubMenu';
+import Switch from '@material-ui/core/Switch';
+import Box from '@material-ui/core/Box';
+import useTheme from '@material-ui/styles/useTheme';
+
+export default function CascadingMenu() {
+  const theme = useTheme();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [darkMode, setDarkMode] = useState(false);
+
+  const handleButtonClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const swapDarkMode = () => {
+    setDarkMode(!darkMode);
+  };
+
+  return (
+    <div>
+      <Button
+        aria-controls="cascading-menu"
+        aria-haspopup="true"
+        onClick={handleButtonClick}
+      >
+        Open Menu
+      </Button>
+      <Menu
+        id="cascading-menu"
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'top',
+          horizontal: theme.direction === 'rtl' ? 'left' : 'right',
+        }}
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+      >
+        <MenuItem
+          subMenu={
+            <SubMenu>
+              <MenuItem onClick={swapDarkMode}>
+                Dark Mode
+                <Box ml={1}>
+                  <Switch size="small" checked={darkMode} />
```

Comment:
Since this pattern has some a11y issues and we already have a global dark-mode switch we should remove any nested controls for now. We plan on adding separate components to make them fully accessible but until then let's not spread problematic patterns.

Reply:
Cool, I'll update the demo to remove that.